sycl/doc/extensions/experimental/sycl_ext_intel_usm_address_spaces.asciidoc

```diff
@@ -0,0 +1,169 @@
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+== Modifications to SYCL 2020
+
+The following sections contain the related changes and additions to the SYCL
+2020 specification relating to this extension.
+
+=== SYCL Device Memory Model
+
+Add to the end of the definition of global memory:
+Global memory is a virtual address space which overlaps the device and host
+address spaces.
+
+Add two new memory regions as follows:
+
+*Device memory* is a sub-region of global memory that is not directly accessible
+by the host.  Global accessors and USM allocations of the device alloc type
+reside in this address space.
+
+*Host memory* is a sub-region of global memory.  USM pointers allocated with the
+host alloc type reside in this address space.
+
+
+=== Multi-pointer Class
+
+Add the following enumerations to the `access::address_space` enum:
+```c++
+enum class address_space : /* unspecified */ {
+  ...
+  ext_intel_global_device_space,
+  ext_intel_global_host_space
+};
+```
+
+Add the following new conversion operator to the `multi_ptr` class:
+```c++
+// Explicit conversion to global_space
+// Only available if Space == address_space::ext_intel_global_device_space || Space == address_space::ext_intel_global_host_space
+explicit operator multi_ptr<ElementType, access::address_space::global_space, DecorateAddress>() const;
+```
+
+Change the `multi_ptr` constructor taking an accessor with `target::device` to
+also allow `access::address_space::ext_intel_global_device_space` as follows:
+
+--
+[options="header"]
+|===
+| Constructor | Description
+a|
+```c++
+template <int Dimensions, access_mode Mode, access::placeholder IsPlaceholder>
+multi_ptr(
+    accessor<ElementType, Dimensions, Mode, target::device, IsPlaceholder>);
+```
+| Available only when:
+`Space == access::address_space::global_space \|\| Space == access::address_space::ext_intel_global_device_space \|\| Space == access::address_space::generic_space`.
+
+Constructs a `multi_ptr` from an accessor of `target::device`.
+
+This constructor may only be called from within a command.
+|===
+--
+
+
+=== Explicit Pointer Aliases
+
+Add `device_ptr` and `host_ptr` aliases to the list of `multi_ptr` aliases as
+follows:
+```c++
+template<typename ElementType, access::decorated IsDecorated = access::decorated::legacy>
+using device_ptr = multi_ptr<ElementType, access::address_space::ext_intel_global_device_space, IsDecorated>
+
+template<typename ElementType, access::decorated IsDecorated = access::decorated::legacy>
+using host_ptr = multi_ptr<ElementType, access::address_space::ext_intel_global_host_space, IsDecorated>
+```
```

Comment:
Clarify which namespace these are in.  I think they are currently in `sycl`, but that's not the right place.  They should either be in `sycl::ext::intel` or `sycl::ext::intel::experimental`, depending on the status of this extension.

I think we should deprecate the existing aliases and add new ones in the correct namespace.

Reply:
Currently it is in the `sycl` namespace, but I agree they should be moved. Should we also add `raw_` and `decorated_` variants while we are in here? @GarveyJoe - Are you okay if we just move them or would you prefer we keep the old ones in the `sycl` namespace as deprecated for now?

Reply:
Please keep the old ones working with deprecation warnings for now so that we can migrate gradually.  FYI, @ajaykumarkannan